packages/hoppscotch-backend/src/auth/guards/github-sso.guard.ts

```diff
@@ -1,8 +1,19 @@
-import { ExecutionContext, Injectable } from '@nestjs/common';
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
 import { AuthGuard } from '@nestjs/passport';
+import { AuthProvider, authProviderCheck, throwHTTPErr } from '../helper';
+import { Observable } from 'rxjs';
 
 @Injectable()
-export class GithubSSOGuard extends AuthGuard('github') {
+export class GithubSSOGuard extends AuthGuard('github') implements CanActivate {
+  canActivate(
+    context: ExecutionContext,
+  ): boolean | Promise<boolean> | Observable<boolean> {
+    if (!authProviderCheck(AuthProvider.GITHUB))
+      throwHTTPErr({ message: 'GitHub auth is not enabled', statusCode: 404 });
```

Comment:
```suggestion
      throwHTTPErr({ message: 'AUTH_PROVIDER_NOT_ENABLED', statusCode: 404 });
```

Reply:
or it can even be `GITHUB_PROVIDER_NOT_ENABLED`

Reply:
I think `AUTH_PROVIDER_NOT_ENABLED` is good